llvm/include/llvm/Analysis/ValueTracking.h

```diff
@@ -791,6 +791,10 @@ bool onlyUsedByLifetimeMarkers(const Value *V);
 /// droppable instructions.
 bool onlyUsedByLifetimeMarkersOrDroppableInsts(const Value *V);
 
+/// Return true if the instruction doesn't potentially cross vector lanes. This
+/// is useful to make GVN-replacements for vector types.
```

Comment:
```suggestion
/// is useful when optimizing based on vector conditions.
```
Though TBH I'd just leave off the second sentence, I don't think it is helpful without a significantly more detailed explanation.